imageio/imageio-bmp/src/main/java/com/twelvemonkeys/imageio/plugins/bmp/BMPImageReader.java

```diff
@@ -196,7 +197,7 @@ public int getHeight(int pImageIndex) throws IOException {
         checkBounds(pImageIndex);
 
         // TODO: Better implementation, include INT_RGB types for 3BYTE_BGR and 4BYTE_ABGR for INT_ARGB
-        return Arrays.asList(getRawImageType(pImageIndex)).iterator();
+        return Collections.singletonList(getRawImageType(pImageIndex)).iterator();
```

Comment:
Since the return of `getRawImageType` is but a single item, a singletonlist can be used.

As the documentation states: `Returns an immutable list containing only the specified object.`

Should be smaller in memory than the default arraylist. It's apparently used a few more times along the codebase.

Reply:
The TODO above the changed line indicates the reason for using a mutable list here... But of course, we could change this until we actually return multiple values.

Reply:
Right. Change removed.